### PR TITLE
Improved docs slightly around default_options

### PR DESCRIPTION
**Pull request by tstone**
Background
-----

Improved docs slightly around `default_options` based on some questions that came up around the first service integration.

Tasks
-----
* [x] [Code of Conduct](https://github.com/Ibotta/atomic_cache/blob/main/CODE_OF_CONDUCT.md) reviewed

### CODE REVIEW

File: docs/PROJECT_SETUP.md [hunk 1]
```diff
@@ -23,9 +23,10 @@ AtomicCache::DefaultConfig.configure do |config|
   config.metrics = Datadog::Statsd.new('localhost', 8125, namespace: 'cache.atomic')
 
   # note: these values can also be set in an env file for env-specific settings
-  config.namespace      = 'atom'
-  config.cache_storage  = AtomicCache::Storage::SharedMemory.new
-  config.key_storage    = AtomicCache::Storage::SharedMemory.new
+  config.namespace        = 'atom'
+  config.default_options  = { generate_ttl_ms: 500 }
+  config.cache_storage    = AtomicCache::Storage::SharedMemory.new
+  config.key_storage      = AtomicCache::Storage::SharedMemory.new
```
Comment (lebeerman) [inline docs/PROJECT_SETUP.md:29]: Being new to gem dev... where's the line in the sand for documentation? Should consumers in the RoR community go to the spec files to see examples for how some of the options are use? 

Would it be for the gem to provide an example for how a Rails app may have `config.dalli_store/memcache_store` options that would be passed into a dalli client or memory instance?

Similarly - is there value in showing an example of a `atomic_cache.fetch` call with options overriding defaults?

Comment (lebeerman) [reply docs/PROJECT_SETUP.md:29]: I dont mean to say our way is THE way but eg:
the note above  `note: these values can also be set in an env file for env-specific settings`
```
  if Rails.env.development? || Rails.env.test?
    config.cache_storage  = AtomicCache::Storage::SharedMemory.new
    config.key_storage    = AtomicCache::Storage::SharedMemory.new
  elsif Rails.env.staging? || Rails.env.production?
    memcache_hosts = Ibotta::Application.config.memcache_hosts
    options = Ibotta::Application.config.cache_store_options

    dc = Dalli::Client.new(memcache_hosts, options)
    config.cache_storage  = AtomicCache::Storage::Dalli.new(dc)
    config.key_storage    = AtomicCache::Storage::Dalli.new(dc)
  end
```

Comment (onyxraven) [reply docs/PROJECT_SETUP.md:29]: good question. You could have a specific rails example for this maybe (leaving out the Ibotta prefixes ;) )

Comment (tstone) [reply docs/PROJECT_SETUP.md:29]: Being an open source project, this is something I've struggled a bit with, is to find a level of documentation that's helpful but not overly Ibotta-specific. I'll try and roll what you have up there into something this morning, and see if we can find a happy middle ground.

Comment (tstone) [reply docs/PROJECT_SETUP.md:29]: Added a `More Complex Rails Configuration` section down below. Have a look. Resolving this conversation.

File: docs/PROJECT_SETUP.md [hunk 2]
```diff
@@ -45,6 +46,49 @@ Note that `Datadog::Statsd` is not _required_.  Adding it, however, will enable
 #### ★ Best Practice ★
 Keep the global namespace short.  For example, memcached has a limit of 250 characters for key length.
 
+#### More Complex Rails Configuration
+
+In any real-world project, the need to run multiple caching strategies or setups is likely to arise. In those cases, it's often advantageous
+to keep a DRY setup, with multiple caching clients sharing the same config. Because Rails initializers run after the environment-specific
+config files, a sane way to manage this is to keep client network settings int he config files, then reference them from the initializer.
```
Comment (lebeerman) [inline docs/PROJECT_SETUP.md:53]: 👍 Thanks for including this! If others use this gem, I'd anticipate this being really helpful.

**Review by lebeerman — APPROVED** [review]

File: docs/PROJECT_SETUP.md (hunk 2 above)
Comment (lebeerman) [reply docs/PROJECT_SETUP.md:53]: Oh and `# config/environments/staging` on #L64 -> `production`

**Pull request merged**